src/game_logic/enemies/eyeball_thrower.hpp

```diff
@@ -24,37 +24,31 @@
 namespace rigel::game_logic::behaviors
 {
 
-namespace eyeball_thrower
-{
-
-struct GettingUp
-{
-  int mFramesElapsed = 0;
-};
-
-struct Walking
-{
-};
-
-struct Attacking
+struct EyeballThrower
 {
-  int mFramesElapsed = 0;
-};
+  struct GettingUp
+  {
+    int mFramesElapsed = 0;
+  };
 
-using State = std::variant<GettingUp, Walking, Attacking>;
+  struct Walking
+  {
+  };
 
-} // namespace eyeball_thrower
+  struct Attacking
+  {
+    int mFramesElapsed = 0;
+  };
 
+  using State = std::variant<GettingUp, Walking, Attacking>;
 
-struct EyeballThrower
-{
   void update(
     GlobalDependencies& dependencies,
     GlobalState& state,
     bool isOnScreen,
     entityx::Entity entity);
 
-  eyeball_thrower::State mState;
+  State mState = GettingUp();
```

Comment:
I do it manually because of weird errors when I leave this variable initializer by default.

`In file included from /root/RigelEngine/src/game_logic/enemies/eyeball_thrower.cpp:17:
/root/RigelEngine/src/game_logic/enemies/eyeball_thrower.hpp:54:16: error: use of deleted function ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::EyeballThrower::GettingUp, rigel::game_logic::behaviors::EyeballThrower::Walking, rigel::game_logic::behaviors::EyeballThrower::Attacking}]’
   54 |   State mState{};
      |                ^
In file included from /root/RigelEngine/src/game_logic/enemies/eyeball_thrower.hpp:21,
                 from /root/RigelEngine/src/game_logic/enemies/eyeball_thrower.cpp:17:
/usr/include/c++/9/variant:1288:7: note: ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::EyeballThrower::GettingUp, rigel::game_logic::behaviors::EyeballThrower::Walking, rigel::game_logic::behaviors::EyeballThrower::Attacking}]’ is implicitly deleted because the default definition would be ill-formed:
 1288 |       variant() = default;
      |       ^~~~~~~
/usr/include/c++/9/variant:1288:7: error: use of deleted function ‘constexpr std::_Enable_default_constructor<false, _Tag>::_Enable_default_constructor() [with _Tag = std::variant<rigel::game_logic::behaviors::EyeballThrower::GettingUp, rigel::game_logic::behaviors::EyeballThrower::Walking, rigel::game_logic::behaviors::EyeballThrower::Attacking>]’
In file included from /usr/include/c++/9/optional:43,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/hashtable.h:37,
                 from /usr/include/c++/9/unordered_map:46,
                 from /root/RigelEngine/3rd_party/entityx/entityx/Event.h:17,
                 from /root/RigelEngine/3rd_party/entityx/entityx/entityx.h:4,
                 from /root/RigelEngine/src/game_logic/global_dependencies.hpp:27,
                 from /root/RigelEngine/src/game_logic/enemies/eyeball_thrower.hpp:19,
                 from /root/RigelEngine/src/game_logic/enemies/eyeball_thrower.cpp:17:
/usr/include/c++/9/bits/enable_special_members.h:110:15: note: declared here
  110 |     constexpr _Enable_default_constructor() noexcept = delete;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/rigel_core.dir/build.make:739: src/CMakeFiles/rigel_core.dir/game_logic/enemies/eyeball_thrower.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:481: src/CMakeFiles/rigel_core.dir/all] Error 2
make: *** [Makefile:152: all] Error 2`

Reply:
Interesting, is that with gcc?